src/m365/spo/commands/homesite/homesite-set.ts

```diff
@@ -77,26 +70,151 @@ class SpoHomeSiteSetCommand extends SpoCommand {
           'content-Type': 'application/json'
         },
         responseType: 'json',
-        data: {
-          sphSiteUrl: args.options.siteUrl
-        }
+        data: {}
       };
 
-      if (args.options.vivaConnectionsDefaultStart !== undefined) {
-        requestOptions.url += '/SetSPHSiteWithConfiguration';
-        requestOptions.data.configuration = { vivaConnectionsDefaultStart: args.options.vivaConnectionsDefaultStart };
+      const homeSiteCount = await this.getHomeSiteCount(spoAdminUrl, logger);
+
+      if (args.options.vivaConnectionsDefaultStart !== undefined ||
+        args.options.draftMode !== undefined ||
+        args.options.audienceIds !== undefined ||
+        args.options.audienceNames !== undefined ||
+        args.options.targetedLicenseType !== undefined ||
+        args.options.order !== undefined) {
+
+        const configuration: any = {};
+        if (args.options.vivaConnectionsDefaultStart !== undefined) {
+          configuration.IsVivaConnectionsDefaultStartPresent = true;
+          configuration.vivaConnectionsDefaultStart = args.options.vivaConnectionsDefaultStart;
+        }
+        if (args.options.draftMode !== undefined) {
+          configuration.IsInDraftModePresent = true;
+          configuration.isInDraftMode = args.options.draftMode;
+        }
+        if (args.options.audienceIds !== undefined) {
+          configuration.IsAudiencesPresent = true;
+          configuration.Audiences = args.options.audienceIds.split(',').map(id => id.trim());
+        }
+        if (args.options.audienceNames !== undefined) {
+          configuration.IsAudiencesPresent = true;
+          configuration.Audiences = await this.transformAudienceNamesToIds(args.options.audienceNames);
+        }
+        if (args.options.targetedLicenseType !== undefined) {
```

Comment:
We shouldn't enhance the legacy code with the new options. Let's leave the code as is.
Also, when specifying this option using the legacy API, I'm getting errors. So let's not try to enhance that request with things it cannot handle.

Reply:
It was a small mistake from my side indeed. But anyway, it was not useful to add these properties to the body of the old API.